presto-hive-common/src/main/java/com/facebook/presto/hive/HiveFileInfo.java

```diff
@@ -30,26 +32,37 @@
     private final long length;
     private final long fileModifiedTime;
     private final Optional<byte[]> extraFileInfo;
+    private final Map<String, String> customSplitInfo;
 
     public static HiveFileInfo createHiveFileInfo(LocatedFileStatus locatedFileStatus, Optional<byte[]> extraFileContext)
+    {
+        return createHiveFileInfo(
+                locatedFileStatus,
+                extraFileContext,
+                ImmutableMap.of());
+    }
+
+    public static HiveFileInfo createHiveFileInfo(LocatedFileStatus locatedFileStatus, Optional<byte[]> extraFileContext, Map<String, String> customSplitInfo)
```

Comment:
Why do we need customSplitInfo in HiveFileInfo ? i thought we found a way to re-use the existing `byte[] extraFileInfo` for our usecase ?

Reply:
We will still need to inject custom property (mutation enabled = true) in the split for worker to deserialize the byte[] accordingly.